store/src/main/java/com/dropbox/android/external/store4/impl/FetcherController.kt

```diff
@@ -65,12 +67,23 @@ internal class FetcherController<Key, Input, Output>(
                 scope = scope,
                 bufferSize = 0,
                 source = flow { emitAll(realFetcher(key)) }.map {
-                    StoreResponse.Data(
-                        it,
-                        origin = ResponseOrigin.Fetcher
-                    ) as StoreResponse<Input>
-                }.catch {
-                    emit(StoreResponse.Error(it, origin = ResponseOrigin.Fetcher))
+                    when (it) {
+                        is FetcherResult.Data -> StoreResponse.Data(
+                            it.value,
+                            origin = ResponseOrigin.Fetcher
+                        ) as StoreResponse<Input>
+                        is FetcherResult.Error.Message -> StoreResponse.Error.Message(
+                            it.message,
+                            origin = ResponseOrigin.Fetcher
+                        )
+                        is FetcherResult.Error.Exception -> StoreResponse.Error.Exception(
+                            it.error,
+                            origin = ResponseOrigin.Fetcher
+                        )
+                    }
+                // }.catch {
+                //     // Should we remove this?
```

Comment:
I feel like yes. thoughts?

Reply:
yes.
actually i added a test related to this such that a fetcher that returns Flow<FetcherResponse> cannot throw, if it throws, it should just crash the application. we should not be catching those.

Can you carry over these tests into your CL?

https://github.com/dropbox/Store/pull/139/files#diff-851fab4fbb224af0e203ae3b5462db40